src/openassetio-core/include/openassetio/hostApi/Manager.hpp

```diff
@@ -498,9 +499,153 @@ class OPENASSETIO_CORE_EXPORT Manager {
+   * entity references. Only traits applicable to the supplied entity
+   * references will be set in the resulting data.
+   *
+   * @param context The calling context.
+   *
+   * @param errorPolicyTag  Parameter for selecting the appropriate
+   * overload (tagged dispatch idiom).
+   *
+   * @return Populated data.
+   */
+  TraitsDataPtr resolve(const EntityReference& entityReference, const trait::TraitSet& traitSet,
+                        const ContextConstPtr& context,
+                        BatchElementErrorPolicyTag::Exception errorPolicyTag = {});
+
+  /**
+   * Provides either a populated @fqref{TraitsData} "TraitsData" or a
+   * @fqref{BatchElementError} "BatchElementError".
+   *
+   * If successful, the result is populated with the
+   * available data for the requested set of traits for the given @ref
+   * entity_reference.
+   *
+   * Otherwise, the result is populated with an error object detailing
+   * the reason for the failure to resolve this particular entity.
+   *
+   * Errors that are not specific to the entity being resolved will be
+   * thrown as an exception.
+   *
+   * @see @ref resolve(const EntityReferences&, <!--
+   * --> const trait::TraitSet&, const ContextConstPtr&, <!--
+   * --> const ResolveSuccessCallback&, <!--
+   * --> const BatchElementErrorCallback& errorCallback)
+   * "Callback signature" variation for more details.
+   *
+   * @param entityReference Entity reference to query.
+   *
+   * @param traitSet The trait IDs to resolve for the supplied list of
+   * entity references. Only traits applicable to the supplied entity
+   * references will be set in the resulting data.
+   *
+   * @param context The calling context.
+   *
+   * @param errorPolicyTag  Parameter for selecting the appropriate
+   * overload (tagged dispatch idiom).
+   *
+   * @return Object containing either the populated data or an error
+   * object.
+   */
+  std::variant<TraitsDataPtr, BatchElementError> resolve(
+      const EntityReference& entityReference, const trait::TraitSet& traitSet,
+      const ContextConstPtr& context, BatchElementErrorPolicyTag::Variant errorPolicyTag);
+
+  /**
+   * Provides a @fqref{TraitsData} "TraitsData" populated with the
+   * available data for the requested set of traits for each given @ref
+   * entity_reference.
+   *
+   * @see @ref resolve(const EntityReferences&, <!--
+   * --> const trait::TraitSet&, const ContextConstPtr&, <!--
+   * --> const ResolveSuccessCallback&, <!--
+   * --> const BatchElementErrorCallback& errorCallback)
+   * "Callback signature" variation for more details.
+   *
+   * Any errors that occur during resolution will be immediately thrown
+   * as an exception, either from the @ref manager plugin (for errors
+   * not specific to the entity reference) or as a
+   * @fqref{BatchElementException} "BatchElementException"-derived
+   * error. This may interrupt the resolution of entity references that
```

Comment:
I suppose it's `will` at the moment yeah. It's `may` (or rather, a more expanded explanation), once the failfast option is added.

Reply:
Depends on the definition of "processed".  If that refers to the manager, then the manager might defer calling the callback until it's finished the whole batch (or maybe a chunk of the batch, or whatever); or it might be multithreading.  

That's what I had in mind when I originally wrote that docstring, anyway, so the caller knows that the backend system might still be hit with lots of queries.

Reply:
Right, got you - We're generally only exposing the work done by the manager in a loose fashion, in terms of is it trivial or not - given that it is so implementation specific. So maybe not worth mentioning unless it has an impact on the subsequent control flow or execution time of the hosts code?

Reply:
https://github.com/OpenAssetIO/OpenAssetIO/pull/864/commits/5dd8db58b4bc740584fa8cfca6f388b45dc7deb1

As discussed, just removed this comment.